api-report/container-runtime.api.md

```diff
@@ -147,7 +147,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     // Warning: (ae-forgotten-export) The symbol "GCNodeType" needs to be exported by the entry point index.d.ts
     getNodeType(nodePath: string): GCNodeType;
     // (undocumented)
-    getPendingLocalState(): unknown;
+    getPendingLocalState(props?: {
+        waitBlobsToAttach: boolean;
```

Comment:
in our definitional layers I think it would like use to be less coupled to blob, as the api isn't blob specific above the container runtime. I found this [random stack exchange entry about linux kernel shutdown](https://unix.stackexchange.com/questions/122557/how-does-the-system-shutdown-of-a-linux-kernel-work-internally) for inspiration. i find it help to see how other systems do similar things.  What i found interesting there in this context was the `reboot_notifier_list` as it seems similar in intent to what we are trying to do, specifically notify the layers that a close is coming. 

In fluid we don't have the concepts of `shutdown` or `reboot`, so i don't want to introduce those, but what we do have is `close`. This gets me to something like `notifyClose`, but i'd like it to be a bit more explicit in that we aren't actually closed yet, just that we will be soon, which got me to `notifyPendingClose`. however, i didn't like the reuse of pending here, as we are already using it when talking about pending state,  i don't want to confuse the two. so i just searched for the word pending and got some synonyms, i think the one i like most is `imminent` as i think it captures that its going to happen very soon, and can't be stopped, this lands me at `notifyImminentClosure`